### DISCUSSION TITLE
SnapKey Not Working with Valorant on Any Key (SnapKey v1.2.1)

### DISCUSSION THREAD

**Discussion by happychristian777** [cafali/SnapKey#64] [General]
I'm experiencing an issue with the Snap Key feature in Snap Tap version 1.2.1. When using the Snap Key with Valorant, the functionality doesn't work regardless of the key I bind it to. This issue persists across all key bindings.

Steps to Reproduce:

    Open Snap Tap (v1.2.1).
    Set up a Snap Key for use with Valorant.
    Attempt to use the Snap Key in-game with any key binding.
    Observe that the Snap Key does not function as expected.

Expected Behavior:
The Snap Key should trigger the assigned action in Valorant when the bound key is pressed.

Actual Behavior:
The Snap Key does not trigger any action in Valorant, regardless of the key binding.

Additional Information:

    Snap Tap Version: v1.2.1
    Operating System: windows 11

**Comment by cafali** [comment]
Tested Valo on my Setup - Works just fine 

It sounds like the issue may be related to your setup or keyboard layout, rather than SnapKey itself. Could you please provide more information, such as:

- Your keyboard + keyboard layout 
- Please provide a screenshot or  copy the content of your current `config.cfg` file.
```
config.cfg

[Group]
key1=65
key2=68

[Group]
key3=83
key4=87
```
- Have you tested SnapKey in other games? If so, what were the results?
- Please test SnapKey's functionality (WASD by default) using this tool: https://drakeirving.github.io/MultiKeyDisplay/
- Ensure that the KeyCodes are correct by checking them here: https://www.toptal.com/developers/keycode

Also, check out the SnapKey Wiki and the Rebinding Keys guide here: https://github.com/cafali/SnapKey/wiki/Rebinding-Keys

Additional Steps:

- Make sure SnapKey.exe and config.cfg are in the same folder.
- Avoid putting SnapKey in a folder that needs Admin privileges, as it might not save the changes you make to the cfg file, if SnapKey is in a folder with restricted access, it could prevent you from making and saving any updates to the configuration file e.g **C:\Program Files and C:\Program Files (x86)**

**Comment by herovaxy** [comment]
If you've tried other games and the SnapKey is actually working fine. Then it's clear that it's how Valorant movement works in the game.

I've tried using AHK scripts before, and I quickly noticed that SnapTap doesn't work in Valorant, as counter-strafing behaves the same with or without SnapTap enabled.

However, if you hold the A key and then press D, you'll notice that the counter-strafing becomes instant. This seems to be just how Valorant handles movement. Unlike CS2, the model will counter strafe instantly the moment you pressed the opposite key.

I suggest you try test this scenario on both CS2 and Valorant without SnapKey. Try perform the counter strafe and you'll see that CS2 strafe much more faster than Valorant does.

**Comment by cafali** [comment]
> If you've tried other games and the SnapKey is actually working fine. Then it's clear that it's how Valorant movement works in the game.
> 
> I've tried using AHK scripts before, and I quickly noticed that SnapTap doesn't work in Valorant, as counter-strafing behaves the same with or without SnapTap enabled.
> 
> However, if you hold the A key and then press D, you'll notice that the counter-strafing becomes instant. This seems to be just how Valorant handles movement. Unlike CS2, the model will counter strafe instantly the moment you pressed the opposite key.
> 
> I suggest you try test this scenario on both CS2 and Valorant without SnapKey. Try perform the counter strafe and you'll see that CS2 strafe much more faster than Valorant does.

I think what @happychristian777 means is that SnapKey is not canceling the last pressed key when a new key is pressed, as it should. The SnapKey functionality isn't working for him regardless of Valorant movement; the functionality should be there.

**Comment by happychristian777** [comment]
Hi,
so my keyboard layout is 75% (84-key) keychron k2 v2
screenshot: config.cfg
[image: image.png]
i've tested it:https://drakeirving.github.io/MultiKeyDisplay/ and it works
fine here
maybe valorant is blocking SnapKey

**Comment by minteeaa** [comment]
unable to replicate on SnapKey 1.2.1 and a Tofu60 - nothing in Valorant patch notes about blocking this feature/program either; though CS2 devs have added a check for this feature

> screenshot: config.cfg
> [image: image.png]

please fix your image attachment and double check if your config is set-up properly; though if it works in MultiKeyDisplay, this could be purely a visual thing as commented above:

> I've tried using AHK scripts before, and I quickly noticed that SnapTap doesn't work in Valorant, as counter-strafing behaves the same with or without SnapTap enabled.
> However, if you hold the A key and then press D, you'll notice that the counter-strafing becomes instant. This seems to be just how Valorant handles movement. Unlike CS2, the model will counter strafe instantly the moment you pressed the opposite key.

**Comment by minteeaa** [comment]
on a secondary note, if the keys themselves stop working entirely, then we might have a situation of Valorant blocking synthetic inputs - if you can please be a little more specific as to _what isn't working_

**Comment by happychristian777** [comment]
There is no effect after turning on of snap key with valorant i’ve tested
it on other tools, it works fine but not in valorant so maybe vanguard or
something blocking your app

**Comment by minteeaa** [comment]
can't confirm without proper demonstration and so far nobody else within this issue report has been able to replicate this behavior - on this note we'll keep an eye on Vanguard compatibility (Riot is fairly sneaky; wouldn't be surprised) but this cannot be listed as a bug for the time being

**Comment by cafali** [comment]
We’ll move your issue to discussions to get more feedback from others since we couldn’t recreate the problem and it seems to be working fine for others.

**Comment by Shirtpeas** [comment]
I can also confirm that when testing on https://drakeirving.github.io/MultiKeyDisplay/ Snapkey appears to be working (eg. holding down A and tapping D swaps the inputs), but this does not work in the games I've tested (CS2 and Apex). 

This can also be demonstrated just typing in Notepad++ and then trying the same in CS2 console: In Notepad++, holding A will type the letter 'a', then holding D will type 'd', and when releasing D it will again type the letter 'a'.  In the CS2 console, holding down A will spam the letter 'a' over and over (as per normal behaviour without Snapkey enabled, at least in my case), then when hitting D it will spam 'd', but not revert to 'a' when released.

If this is working on a hardware/Windows API level I find it baffling that it works in Windows but not in the games I've tested.

**Reply by minteeaa** [reply]
CS2 devs have added a check for the "snap tap" feature already - it hasnt been made public for what that is used for but it could be the beginning of blocking this type of thing;

it's very odd that it would happen so widely all of a sudden though, makes me believe it may just be a bug

i can't confirm a bug that i can't replicate, but will continue to monitor this topic for the time being

just to confirm, you are using the latest release of SnapKey? 
is there any other software running that controls the keyboard inputs?

**Reply by Shirtpeas** [reply]
I am using the latest (1.2.1),and I've also tried the previous version.

AFAIK the only software I have installed that affects keyboard inputs is Syanpse.  All that does is sets the actuation point and enables Rapid Trigger on my Razer Huntman Analogue. I've tried disabling that to no effect.

**Reply by Shirtpeas** [reply]
I may have figured it out: I ran Snapkeys as administrator and it immediately started working in CS2. I am runnning steam as admin and therefore it elevates any game launched via steam. Maybe Snapkeys just needs to be equal privelages to whichever game you're playing?

**Reply by Yaw-Dev** [reply]
I've experienced the same thing but I don't think the reason is that it "just needs to be equal privelages to whichever game you're playing". Basically just keep on running it as admin and it should work.

**Reply by minteeaa** [reply]
> I may have figured it out: I ran Snapkeys as administrator and it immediately started working in CS2. I am runnning steam as admin and therefore it elevates any game launched via steam. Maybe Snapkeys just needs to be equal privelages to whichever game you're playing?

as a general note, you shouldn't run Steam or most games as admin - though I know this does fix some problems with certain games

SnapKey is run in user space, so anything with administrator, or "elevated" privileges will not allow SnapKey to input unless you are also running it as administrator

this is the issue here, and the proper fix for this at noted above is just to run as administrator - good catch

**Comment by happychristian777** [comment]
It worked after i run it as admin with valorant
